sale_order_back2draft/models/sale_order.py

```diff
@@ -0,0 +1,24 @@
+# -*- encoding: utf-8 -*-
+##############################################################################
+# For copyright and license notices, see __openerp__.py file in root directory
+##############################################################################
+
+from openerp import models, api, exceptions, _
+
+
+class SaleOrder(models.Model):
+    _inherit = "sale.order"
+
+    @api.multi
+    def button_draft(self):
+        # go from canceled state to draft state
+        for order in self:
+            if order.state != 'cancel':
+                raise exceptions.Warning(
+                    _("You can't back any order that it's not on cancel "
+                      "state"))
+            order.order_line.write({'state': 'draft'})
+            order.write({'state': 'draft'})
```

Comment:
Well, this is to assure that workflow is not changed by the state value, and it's not too much performance cost (this is usually called for one record).